chia/server/reconnect_task.py

```diff
@@ -18,6 +18,7 @@ async def connection_check():
                 if connection.get_peer_info() == peer_info or connection.get_peer_info() == peer_info_arg:
                     peer_retry = False
             if peer_retry:
+                peer_info = PeerInfo(socket.gethostbyname(peer_info_arg.host), peer_info_arg.port)
```

Comment:
is this an async call? It looks like this could block as it is now.

Reply:
Yes, I think you are right, it might block if the lookup takes significant time.
I guess the only way to avoid that would be to wrap the lookup in an executor call or use an asyncronous DNS library :/
For a harvester for example, the blocking might not be a problem because it cannot really do anything if it's not connected to it's one peer anyway, the same might not be true for other services with multiple peers. Maybe it could be changed for some services only?